smartsim/_core/_cli/__main__.py

```diff
@@ -32,7 +32,11 @@
 def main() -> int:
     smart_cli = default_cli()
 
-    return smart_cli.execute(sys.argv)
+    try:
+        return smart_cli.execute(sys.argv)
+    except KeyboardInterrupt:
+        print("Dashboard terminated by user")
```

Comment:
Are we wrapping our whole CLI in this ``try/except`` block? If so, the error message seems a bit hyper specific. For instance in the following case:
```py
$ smart build --device cpu
# some build output here
^C  # Yikes! I actually wanted ``--device gpu``, better sigterm!
Dashboard terminated by user  # I, as a user, am now confused
                              # "I didn't start a dashboard? I'm just trying to build SS??"

Reply:
Repaired the obviously wrong messages by adding a fallthrough if a `SmartSimInterrupt` with specific detail isn't caught. This is still not perfect. Ctrl-C can still happen before the entrypoints start and show some stack traces... Needs a better fix later